source/agora/node/Ledger.d

```diff
@@ -798,8 +798,19 @@ public class Ledger
 
             const CR = this.enroll_man.getCommitmentNonce(K, block.header.height);  // commited R
             if (CR == Point.init)
+            {
+                log.error("Block#{}: Validator {} (idx: {}) Couldn't find commitment for this validator",
+                          block.header.height, K, idx);
                 return "Block: Couldn't find commitment for this validator";
-            Point R = CR + challenge.toPoint();
+            }
+            const preimage = this.enroll_man.getValidatorPreimageAt(K, Height(block.header.height - 1)); // Preimage at previous block height
+            if (preimage == Hash.init)
+            {
+                log.error("Block#{}: Validator {} (idx: {}) Couldn't find pre-image for this validator",
+                          block.header.height, K, idx);
+                return "Block: Couldn't find pre-image for this validator";
+            }
```

Comment:
The caller will log the error, so I don't see why we would log it here too ?

Reply:
It will but with less information.